CONTRIBUTING.md

```diff
@@ -1,51 +1,147 @@
+1. Navigate to `C:\Users\YourUsername\AppData\Roaming\Code\User\settings.json` and add the following lines:
 
-`$roslynRepoRoot/artifacts/bin/Microsoft.CodeAnalysis.LanguageServer/Debug/net7.0/Microsoft.CodeAnalysis.LanguageServer.dll`
+```json
+"razor.languageServer.debug": true,
+"razor.languageServer.directory": "$razorRepoRoot\\artifacts\\bin\\rzls\\Debug\\net8.0",
+"razor.server.trace": "Debug"
+```
 
-depending on which configuration is built.  Then, launch the extension here and change the VSCode setting `dotnet.server.path` to point to the Roslyn dll path you built above and restart the language server.
+Replace `$razorRepoRoot` and `YourUsername` with your actual values.
 
-If you need to debug the server, you can set the VSCode setting `dotnet.server.waitForDebugger` to true.  This will trigger a `Debugger.Launch()` on the server side as it starts.
+Or, in VSCode settings (`Ctrl+,`):
 
-### Using a locally developed Razor server
+1. Search for `Razor`.
+2. Set `razor.languageServer.directory` to the path of your Razor DLL.
+3. Enable `razor.languageServer.debug`.
+4. Set `razor.server.trace` to `Debug`. This gives you more detailed log messages in the output window.
 
-https://github.com/dotnet/razor contains the server implementation.  Follow the instructions there to build the repo as normal.  Once built, the server will be located in the build output directory, typically
+### Debugging Local Language Servers
 
-`$razorRepoRoot/artifacts/bin/rzls/Debug/net7.0`
+Before running the language servers, ensure you have followed the steps in the [Configuring Local Language Servers](#configuring-local-language-servers) section to configure either the Roslyn or Razor language servers for debugging.
 
-depending on which configuration is built.  Then, launch the extension here and change the VSCode setting `razor.languageServer.directory` to point to the Razor executable path you built above and reload the window.
+After completing the configuration, follow these steps:
 
-If you need to debug the server, you can set the VSCode setting `razor.languageServer.debug` to true.  This will trigger a `Debugger.Launch()` on the server side as it starts. You can also set `razor.server.trace` to `Debug` to get more log messages in the output window
+1. Ensure the language server is fully built in Debug mode.
+2. Open the `.sln` solution file for the language server in a Visual Studio instance. Keep this instance open for use in a later step.
+3. Press `Ctrl+Shift+P` and select `Reload Window`. This ensures the changes from the configuration step are applied.
+4. After reloading, a window will pop up prompting you to select or open a Visual Studio instance. Click `Cancel` at this stage. This is because we want to debug the child VS Code instance that will open after the next step, not the current instance.
```

Comment:
Can you clarify this?  At least for roslyn, if you set `waitForDebugger` you should only get 1 window prompt and you should attach to the first one.

Reply:
> Can you clarify this? At least for roslyn, if you set `waitForDebugger` you should only get 1 window prompt and you should attach to the first one.

I think this is because the advice is to set up the local language servers in the main VS Code settings. I would advise against that, and only configure local language servers in the workspaces that you intend to debug, that way your VS Code installation is unaffected.

Reply:
I did a redo of this section, using workspace approach instead